File: Data Structures & Algorithms/max-water-container/submission-9.java
class Solution {
    public int maxArea(int[] heights) {
        int l=0;
        int r= heights.length -1;
        int maxe =0;
        
        while(l<r){
            int hei = Math.min( heights[l], heights[r]);
            int breath = r-l;
            maxe = Math.max(maxe,hei*breath);
            if( heights[l]< heights[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return maxe;
        
        
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-9.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        map={}
        j=0
        maxe =0
        maxlen =0
        for i in range (len(s)):
            map[s[i]] = map.get(s[i],0)+1
            maxe = max(maxe,map[s[i]]) 
            while ((i-j+1)-maxe) > k:
                map[s[j]]-=1
                j+=1
            maxlen = max(maxlen,i-j+1)
        return maxlen



        